charts/tempo/values.yaml

```diff
@@ -118,39 +120,57 @@ tempo:
 # -- Tempo configuration file contents
 # @default -- Dynamically generated tempo configmap
 config: |
-    multitenancy_enabled: {{ .Values.tempo.multitenancyEnabled }}
-    search_enabled: {{ .Values.tempo.searchEnabled }}
-    usage_report:
-      reporting_enabled: {{ .Values.tempo.reportingEnabled }}
-    metrics_generator_enabled: {{ .Values.tempo.metricsGenerator.enabled }}
-    compactor:
-      compaction:
-        block_retention: {{ .Values.tempo.retention }}
-    distributor:
-      receivers:
-        {{- toYaml .Values.tempo.receivers | nindent 8 }}
-    ingester:
-      {{- toYaml .Values.tempo.ingester | nindent 6 }}
-    server:
-      {{- toYaml .Values.tempo.server | nindent 6 }}
+  multitenancy_enabled: {{ .Values.tempo.multitenancyEnabled }}
+  search_enabled: {{ .Values.tempo.searchEnabled }}
+  usage_report:
+    reporting_enabled: {{ .Values.tempo.reportingEnabled }}
+  metrics_generator_enabled: {{ .Values.tempo.metricsGenerator.enabled }}
+  compactor:
+    compaction:
+      block_retention: {{ .Values.tempo.retention }}
+  distributor:
+    receivers:
+      {{- toYaml .Values.tempo.receivers | nindent 4 }}
+  {{- with .Values.tempo.ingester }}
+  ingester:
+    {{- toYaml . | nindent 2 }}
+  {{- end }}
+  {{- with .Values.tempo.server }}
+  server:
+    {{- toYaml . | nindent 2 }}
+  {{- end }}
+  {{- with .Values.tempo.storage }}
+  storage:
+    {{- toYaml . | nindent 2 }}
+  {{- end }}
+  {{- with .Values.tempo.querier }}
+  querier:
+    {{- toYaml . | nindent 2 }}
+  {{- end }}
+  {{- with .Values.tempo.queryFrontend }}
+  query_frontend:
+    {{- toYaml . | nindent 2 }}
+  {{- end }}
+  {{- if or .Values.tempo.global_overrides .Values.tempo.metricsGenerator.enabled }}
+  overrides:
+    {{- with .Values.tempo.global_overrides }}
+    {{- toYaml . | nindent 2 }}
+    {{- end }}
+    {{- if .Values.tempo.metricsGenerator.enabled }}
+    metrics_generator_processors:
+      - 'service-graphs'
+      - 'span-metrics'
+  metrics_generator:
```

Comment:
it's weird that this is inside of the is statement on line 154. maybe put this in it's own "if metrics generator enabled" condition?